src/components/dex/Chart.vue

```diff
@@ -494,6 +517,11 @@ export default {
     loadChartPromise.then(() => {
       this.loadChart();
     });
+
+    this.saveChartStateInterval = setInterval(() => {
+      this.saveChartState();
+    }, this.$constants.timeouts.DEX_SAVE_CHART_STATE);
```

Comment:
Shouldn’t there be a simple way the state can be watched for change and just write on state changes Instead of polling to achieve this?

Reply:
Yeah, made a note re that: Attempted to save the state based on user action / the chart state being amended / a tradingview method, however after a couple of hours of documentation reading and testing solutions, I couldn't implement this approach so used an interval instead. This does the job and will satisfy users.

Reply:
Looked around the web and [here](https://github.com/mmmy/css3demos/wiki/Widget-Methods), version we have is different. Tried onIntervalChange, ours is called onIntervalChanged but didn't seem to work. I know the polling solution is naff but best I could do :/

Reply:
Ok, perhaps add a comment note or todo so it can possibly be revisited in the future.

Reply:
Have done so in an additional [commit](https://github.com/aphtoken/aph-desktop-web-dex/commit/b647b7086cb3c3f51578fe50c61c5c4f7f8fb862).